hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -110,17 +120,47 @@ public void enrichEvent(StructuredTrace trace, Event event) {
    * enricher class itself could be renamed to be more specific.
    */
   private void enrichHostHeader(Event event) {
-    for (String key: HOST_HEADER_ATTRIBUTES) {
-      String value = SpanAttributeUtils.getStringAttribute(event, key);
-      if (value != null && !value.isEmpty()) {
-        // Ignore if it's "localhost"
-        String host = sanitizeHostValue(value);
-        if (!LOCALHOST.equalsIgnoreCase(host)) {
-          addEnrichedAttribute(event, HOST_HEADER, AttributeValueCreator.create(host));
-          break;
+    Protocol protocol = EnrichedSpanUtils.getProtocol(event);
+    if (protocol == Protocol.PROTOCOL_GRPC) {
+      Optional<String> value = getGrpcAuthority(event);
+      if (value.isPresent()) {
+        Optional<String> host = getSanitizedHostValue(value.get());
+        if (host.isPresent()) {
+          addEnrichedAttribute(event, HOST_HEADER, AttributeValueCreator.create(host.get()));
+          return;
         }
       }
     }
+    for (String key : HOST_HEADER_ATTRIBUTES) {
+      String value = SpanAttributeUtils.getStringAttribute(event, key);
+      Optional<String> host = getSanitizedHostValue(value);
+      if (host.isPresent()) {
+        addEnrichedAttribute(event, HOST_HEADER, AttributeValueCreator.create(host.get()));
+        return;
+      }
+    }
+  }
+
+  private Optional<String> getGrpcAuthority(Event event) {
+    String authority = null;
+    if (event.getGrpc() != null && event.getGrpc().getRequest() != null) {
+      Request request = event.getGrpc().getRequest();
+      RequestMetadata requestMetadata = request.getRequestMetadata();
+      if (requestMetadata != null) {
+        authority = requestMetadata.getAuthority();
```

Comment:
This can directly call `getSanitizedHostValue()` so that you return the final sanitized value.

Reply:
Done